Tests/BotTests/Interceptor/Coordinator.swift

```diff
@@ -0,0 +1,95 @@
+import Foundation
+
+private final class RecordingContext {
+    var destination: URL
+    var recordedStubs: [Interceptor.Stub] = []
+    var session: URLSession
+
+    init(destination: URL) {
+        self.destination = destination
+        self.session = URLSession(configuration: {
+            let config = URLSessionConfiguration.default
+            config.requestCachePolicy = .reloadIgnoringLocalCacheData
+            config.urlCache = nil
+            return config
+        }())
+    }
+}
+
+final class Coordinator {
+    private var stubs: [Interceptor.Stub] = []
+    private var recordingContext: RecordingContext? = nil
+
+    var isRecordingEnabled: Bool {
+        return recordingContext != nil
+    }
+
+    deinit {
+        Interceptor.stopRecording()
+    }
+
+    func load(stubs newStubs: [Interceptor.Stub]) {
+        stubs.append(contentsOf: newStubs)
+    }
+
+    func loadOrRecordStubs(into url: URL) {
+
+        if FileManager.default.fileExists(atPath: url.absoluteString) {
+
+            guard
+                let data = FileManager.default.contents(atPath: url.absoluteString),
+                let loadedStubs = try? JSONDecoder().decode([Interceptor.Stub].self, from: data)
+                else { fatalError("Failed to decode the existent stubs at `\(url)`") }
+
+            stubs.append(contentsOf: loadedStubs)
+
+        } else {
+            recordingContext = RecordingContext(destination: url)
+        }
+    }
+
+    func stopRecording() {
+        guard let context = recordingContext else { return }
+
+        guard
+            let data = try? JSONEncoder().encode(context.recordedStubs),
+            FileManager.default.createFile(atPath: context.destination.absoluteString, contents: data, attributes: nil)
+            else { fatalError("Failed to record stubs at `\(context.destination)`") }
+
+        recordingContext = nil
+    }
+
+    func stub(for request: URLRequest) -> Interceptor.Stub? {
+        guard stubs.isEmpty == false else { return nil }
+
+        return stubs.remove(at: 0)
```

Comment:
So you don't try to match the `URLRequest`? If it happens that we added the stubs for the wrong URLRequest, we won't see the issue then? (and if it's intended, and you just want to `pop` the next stub regardless of the request, why pass the `request` as parameter?)

Reply:
This should be improved in the near future but I didn't want to extend this PR even more in terms of time and complexity, I can add an issue to ensure we track this after being merged.

Reply:
Could you add a comment mentioning the ticket you created so we don't forget to implement it?
```suggestion
        // TODO: [CNSMR-???] Check against `URLRequest` to see if we're pop-ing the expected stub
        return stubs.remove(at: 0)
```